spark/src/main/scala/org/apache/spark/sql/delta/constraints/DeltaInvariantCheckerExec.scala

```diff
@@ -71,9 +73,16 @@ case class DeltaInvariantCheckerExec(
     if (constraints.isEmpty) return child.execute()
     val invariantChecks =
       DeltaInvariantCheckerExec.buildInvariantChecks(child.output, constraints, session)
-    val boundRefs = invariantChecks.map(_.withBoundReferences(child.output))
+
+    // Resolve current_date()/current_time() expressions.
+    // We resolve currentTime for all invariants together to make sure we use the same timestamp.
+    val invariantsFakePlan = AnalysisHelper.FakeLogicalPlan(invariantChecks, Nil)
+    val newInvariantsPlan = optimizer.ComputeCurrentTime(invariantsFakePlan)
 
     child.execute().mapPartitionsInternal { rows =>
+      val boundRefs = newInvariantsPlan.expressions
```

Comment:
can we put the `boundRefs` variable out of the RDD `mapPartitionsInternal` closure as before?

Reply:
I moved it there because I was having a serialization issue that only appeared in some Spark versions. Are there any issues for having the `boundRefs` there?

Reply:
referencing `child.output` in the RDD closure means serializing the entire `child` plan. We should at least do `val localOutput = child.output` before the `mapPartitionsInternal` call and reference this local variable in the closure.

Reply:
Done thanks!